imap_processing/glows/l0/glows_l0_data.py

```diff
@@ -58,63 +79,70 @@ class HistogramL0:
         Number of events
-    HISTOGRAM_DATA : int
-        List of histogram data values
-    """
-
-    def __repr__(self):
-        """Print the data.
-
-        Returns
-        -------
-        String representation of GlowsHistL0
-        """
-        output = "{"
-        for key in self.packet_keys:
-            output += f"{key}: {getattr(self, key)}, "
-        return output + "}"
-
-    def __init__(self, packet):
-        self.packet_keys = []
-        for key, value in packet.data.items():
-            if key != "HISTOGRAM_DATA":
-                setattr(self, key, value.derived_value)
-            else:
-                setattr(self, key, self._convert_histogram_data(value.raw_value))
+    HISTOGRAM_DATA : bin
+        Raw binary format histogram data
 
-            self.packet_keys.append(key)
-
-    def _convert_histogram_data(self, binary_hist_data: str) -> list[int]:
-        """Convert the raw histogram data into a list.
+    """
 
-        This method converts a binary number into a list of histogram values by
-        splitting up the raw binary value into 8-bit segments.
+    MET: int
+    STARTID: int
+    ENDID: int
+    FLAGS: int
+    SWVER: int
+    SEC: int
+    SUBSEC: int
+    OFFSETSEC: int
+    OFFSETSUBSEC: int
+    GLXSEC: int
+    GLXSUBSEC: int
+    GLXOFFSEC: int
+    GLXOFFSUBSEC: int
+    SPINS: int
+    NBINS: int
+    TEMPAVG: int
+    TEMPVAR: int
+    HVAVG: int
+    HVVAR: int
+    SPAVG: int
+    SPVAR: int
+    ELAVG: int
+    ELVAR: int
+    EVENTS: int
+    HISTOGRAM_DATA: bin
+
+    def __init__(self, packet, software_version: str, packet_file_name: str):
+        """Initialize data class with a packet of histogram data.
 
         Parameters
         ----------
-        binary_hist_data : str
-            Raw data read from the packet, in binary format.
-
-        Returns
-        -------
-        histograms: list[int]
-            List of binned histogram data
+        packet
+            Packet generated from space_packet_parser. Should be a type NamedTuple
+            with header and data fields.
+        software_version: str
+            The version of the ground software being used
+        packet_file_name: str
+            The filename of the packet
         """
-        # Convert the histogram data from a large raw string into a list of 8 bit values
-        histograms = []
-        for i in range(8, len(binary_hist_data), 8):
-            histograms.append(int(binary_hist_data[i - 8 : i], 2))
-
-        if len(histograms) != 3599:
-            raise ValueError(
-                f"Histogram packet is lacking bins. Expected a count of 3599, "
-                f"actually received {len(histograms)}"
-            )
+        super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
 
-        return histograms
+        attributes = [field.name for field in fields(self)]
 
+        # For each item in packet, assign it to the matching attribute in the class.
```

Comment:
Would it make sense to subclass your CcsdsData dataclass for these so you don't have to rewrite this block of code each time? Or make an abstract base class that all of them can utilize and you just extend a few fields on all of them?

Reply:
I considered it, but in the end decided that since CcsdsData is a different structure that isn't really a parent structure of the Level 0 data, it is preferable to keep it as a separate class. I may end up making a base class for all these dataclass structures if they end up having a lot of similar code, so I could be talked into doing that now if you feel strongly about it.